server/src/main/java/org/opensearch/search/startree/filter/DimensionFilterMerger.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.startree.filter;
+
+import org.apache.lucene.util.BytesRef;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DimensionFilterMerger {
+
```

Comment:
Can this be an interface instead ? 

And every instance will implement a way to merge dimension filters. 

- Exact with Exact
- Exact with range
- range with range

and so on in future.

Reply:
Right now I prefer to keep it like this. Moving merge logic into each filter type would scatter the implementation, making it harder to maintain and test I feel. The current approach looks cleaner and easier to extend particularly after making required changes around DimensionFilterMapper. We can revisit this if we need more complex merge operations in the future, but for now, this doesn't seem to be immediate requirement. Wdyt?

Reply:
I meant keeping this class `as is`

Just changing to `interface` instead of a `class` with a static method. 
```
public interface DimensionFilterMerger {

public DimensionFilter intersect(DimensionFilter filter1, DimensionFilter filter2, DimensionFilterMapper mapper);

}
```

And then the sub classes can be
```
class ExactFilterMerger extends DimensionFilterMerger {
}

class ExactRangeFilterMerger extends DimensionFilterMerger {
}

class RangeFiltersMerger extends DimensionFilterMerger {
}
```

Reply:
>We can revisit this if we need more complex merge operations in the future, but for now, this doesn't seem to be immediate requirement. Wdyt?

Since this is a OOP v/s functional debate. I agree that it's not immediate and can naturally evolve into place.

Reply:
Although we should name this class `DimensionFilterMergerUtils` or something to indicate its not an interface like other classes.

Reply:
Done